### DISCUSSION TITLE
Trigger action on specific source branch name

### DISCUSSION THREAD

**Discussion by Benwick91** [community/community#26082] [Actions]
<p>We have a branch model with feature branches, develop, staging (for testing) and master for release. Now we have an action for the Release notes, which creates a Pull Request with the actual release notes when pushing to staging or master.</p>
<p>But the action should only run when we push a Pull Request from develop into staging and staging into master, but not when pushing from the release notes branch into staging or master, because then we always have a new created Pull Request with the release notes. Is this possible? Thank you!</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/benwick91">@benwick91</a>,</p>
<p>You can use <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">on.pull_request.branches</a> to limit the base branch.<br>
And add job level if expression to check ‘github.head_ref’, if it’s not source branch, skip the job.</p>
<p>Code sample as below:</p>
<pre><code class="lang-auto">on:
  pull_request:
    branches:            # limit base branch to staging and master branch.
      - staging
      - master
jobs:
  prcheck:
    if: github.head_ref=='develop' || github.head_ref=='staging'   # check the source branch, skip the job if it's not correct.
    ......
</code></pre>
<p>Thanks</p>
